libbeat/processors/actions/decode_json_fields.go

```diff
@@ -146,6 +146,10 @@ func unmarshal(maxDepth int, text string, fields *interface{}, processArray bool
 			return v, false
 		}
 
+		if !strings.HasPrefix(str, "[") && !strings.HasPrefix(str, "{") {
```

Comment:
You check here for `[` but couldn't find an example of it the tests?

Reply:
I added it, the mentioned bug is about processing when it should not process, so it should be fine to test it with processing enabled

Reply:
It's a quite simple heuristic, but should be fine. It still can fail, though. But decodeJSON should have failed in the first place. For example given the string `"2016-20-10"` the lexer should find that `-` is an invalid character. The `dec.More()` call only returns false (as there is no more valid data in the stream), but `More` also returns `false` if there was an error. We can improve `decodeJSON` by checking the `dec.Token()` error code after `dec.More()`.

For the heuristics (we can still guard decodeJSON using some heuristics), we should remove whitespace when testing as well:

```
func isStructured(s string) bool {
  s = strings.TrimSpace(s)
  end = len(s) - 1
  return end > 0 && (
    (s[0] == '[' && s[end] == ']') ||
    (s[0] == '{' && s[end] == '}')
  )
}
```